model.py

```diff
@@ -6633,6 +6633,69 @@ def persistent_token_create(self, _db, data_source, type, patron):
 # Index to make lookup_by_token() fast.
 Index("ix_credentials_data_source_id_type_token", Credential.data_source_id, Credential.type, Credential.credential, unique=True)
 
+
+class DelegatedPatronIdentifier(Base):
+    """This library is in charge of coming up with, and storing,
+    identifiers associated with the patrons of some other library.
+
+    e.g. NYPL provides Adobe IDs for patrons of all libraries that use
+    the SimplyE app.
+
+    Those identifiers are stored here.
+    """
+    ADOBE_ID = 'Adobe ID'
+    
+    __tablename__ = 'delegatedpatronidentifiers'
+    id = Column(Integer, primary_key=True)
+    type = Column(String(255), index=True)
+    library_uri = Column(String(255), index=True)
+
+    # This is the ID the foreign library gives us when referring to
+    # this patron.
+    patron_identifier = Column(String(255), index=True)
+
+    # This is the identifier we made up for the patron. This is what the
+    # foreign library is trying to look up.
+    delegated_identifier = Column(String)
```

Comment:
We never have to look up a patron based on the delegated identifier. We only look up based on `foreign_identifier`.

To make the example concrete, BPL needs to know a patron's Adobe ID, so they send us the ID they used to register that ID. If they have the Adobe ID there's no need for them to look anything up.